### PR TITLE
MRG: Fix filtering check

### LINKED ISSUE #5614: Epochs filtering error likely due to save/load data typing
raw = mne.io.read_raw_fif(mne.datasets.sample.data_path() + '/MEG/sample/sample_audvis_raw.fif')
events = mne.find_events(raw)
epochs = mne.Epochs(raw,events)
epochs.save('test-epo.fif')
epochs = mne.read_epochs('test-epo.fif')
epochs = epochs.filter(l_freq=None,h_freq=40)

Gives the following error:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 2, in filter
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/utils.py", line 803, in verbose
    return function(*args, **kwargs)
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/filter.py", line 2045, in filter
    fir_window=fir_window, fir_design=fir_design, pad=pad)
  File "<string>", line 2, in filter_data
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/utils.py", line 803, in verbose
    return function(*args, **kwargs)
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/filter.py", line 859, in filter_data
    copy, pad)
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/filter.py", line 164, in _overlap_add_filter
    x, orig_shape, picks = _prep_for_filtering(x, copy, picks)
  File "/autofs/space/karima_001/users/alex/software/anaconda3/lib/python3.6/site-packages/mne-0.17.dev0-py3.6.egg/mne/filter.py", line 269, in _prep_for_filtering
    raise TypeError("Arrays passed for filtering must have a dtype of "
TypeError: Arrays passed for filtering must have a dtype of np.float64

When I look at the dtype of the epochs before being saved it's float64 and after it's dtype('>f8') so I would hazard to guess that this is the issue.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #5614.

@alexrockhill can you verify this fixes your issue?

I also replaced `_TempDir` calls with pytest `tmpdir` fixture, something that I think we can do gradually as we find them.

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -145,6 +145,8 @@ Bug
 
 - Fix problem with :meth:`mne.io.Raw.add_channels` where ``raw.info['bads']`` was replicated by `Eric Larson`_
 
+- Fix bug when trying to filter epochs loaded from disk with :func:`mne.read_epochs` by `Eric Larson`_
```
Comment (agramfort) [inline doc/whats_new.rst]: this is not a bug that was released before no? if it's a bug fix from a PR we just merged we should not put this in what's new

Comment (larsoner) [reply doc/whats_new.rst]: Actually the bug is with `endianness` introduced by that PR, I'll fix that, too, and remove `whats_new.rst` update

File: mne/epochs.py [hunk 2]
```diff
@@ -2580,16 +2580,16 @@ def _read_one_epoch_file(f, tree, preload):
...
-            datatype = np.dtype('>f8')
+            datatype = np.dtype('<f8')
             size_actual = data_tag.size // 8 - 2
         elif data_tag.type == FIFF.FIFFT_COMPLEX_FLOAT:
-            datatype = np.dtype('>c16')
+            datatype = np.dtype('<c16')
             size_actual = data_tag.size // 8 - 2
         elif data_tag.type == FIFF.FIFFT_COMPLEX_DOUBLE:
-            datatype = np.dtype('>c16')
+            datatype = np.dtype('<c16')
```
Comment (larsoner) [inline mne/epochs.py]: These were the bugs

File: mne/filter.py [hunk 3]
```diff
@@ -265,7 +265,7 @@ def _filter_attenuation(h, freq, gain):
 
 def _prep_for_filtering(x, copy, picks=None):
     """Set up array as 2D for filtering ease."""
-    if x.dtype != np.float64:
+    if x.dtype not in ('<f8', '>f8'):
```
Comment (larsoner) [inline mne/filter.py]: ... but having this more general check should still be fine.

**Comment by larsoner** [comment]
... changed my mind, reverted the `mne.filter` change (but made the error message more informative).

In any case, this should fix things.

File: mne/epochs.py [hunk 4]
```diff
@@ -2580,16 +2580,16 @@ def _read_one_epoch_file(f, tree, preload):
         size_expected = len(events) * np.prod(epoch_shape)
         # on read double-precision is always used
         if data_tag.type == FIFF.FIFFT_FLOAT:
-            datatype = np.dtype('>f8')
+            datatype = np.float64
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/epochs.py]: This is used post-reading to do a cast, so no

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**